merge_request_approvals_test.go

```diff
@@ -0,0 +1,153 @@
+			}
+		}
+	],
+	"approver_groups": [
+		{
+			"group": {
+				"id": 1,
+				"name": "group1",
+				"path": "group1",
+				"description": "",
+				"visibility": "public",
+				"lfs_enabled": false,
+				"avatar_url": "https://www.gravatar.com/avatar/4aea8cf834ed91844a2da4ff7ae6b491?s=80\u0026d=identicon",
+				"web_url": "http://localhost/groups/group1",
+				"request_access_enabled": false,
+				"full_name": "group1",
+				"full_path": "group1"
+			}
+		}
+	],
+	"approvals_before_merge": 2,
+	"reset_approvals_on_push": true,
+	"disable_overriding_approvers_per_merge_request": false	
+}`
+
+func exampleParsedResponse() *MergeRequestProjectApprovals {
+	return &MergeRequestProjectApprovals{
+		ApprovalsBeforeMerge:                      2,
+		ResetApprovalsOnPush:                      true,
+		DisableOverridingApproversPerMergeRequest: false,
+		Approvers: []mergeRequestApproverUser{
+			mergeRequestApproverUser{
+				User: struct {
+					ID        int    `json:"id"`
+					Name      string `json:"name"`
+					Username  string `json:"username"`
+					State     string `json:"state"`
+					AvatarURL string `json:"avatar_url"`
+					WebURL    string `json:"web_url"`
+				}{
+					ID:        5,
+					Name:      "John Doe6",
+					Username:  "user5",
+					State:     "active",
+					AvatarURL: "https://www.gravatar.com/avatar/4aea8cf834ed91844a2da4ff7ae6b491?s=80\u0026d=identicon",
+					WebURL:    "http://localhost/user5",
+				},
+			},
+		},
+		ApproverGroups: []mergeRequestApproverGroup{
+			mergeRequestApproverGroup{
+				Group: struct {
+					ID                   int    `json:"id"`
+					Name                 string `json:"name"`
+					Path                 string `json:"path"`
+					Description          string `json:"description"`
+					Visibility           string `json:"visibility"`
+					AvatarURL            string `json:"avatar_url"`
+					WebURL               string `json:"web_url"`
+					FullName             string `json:"full_name"`
+					FullPath             string `json:"full_path"`
+					LFSEnabled           bool   `json:"lfs_enabled"`
+					RequestAccessEnabled bool   `json:"request_access_enabled"`
+				}{
+					ID:                   1,
+					Name:                 "group1",
+					Path:                 "group1",
+					Description:          "",
+					Visibility:           "public",
+					AvatarURL:            "https://www.gravatar.com/avatar/4aea8cf834ed91844a2da4ff7ae6b491?s=80\u0026d=identicon",
+					WebURL:               "http://localhost/groups/group1",
+					FullName:             "group1",
+					FullPath:             "group1",
+					LFSEnabled:           false,
+					RequestAccessEnabled: false,
+				},
+			},
+		},
+	}
+}
+func TestUpdateMergeRequestProjectApprovals(t *testing.T) {
+	mux, server, client := setup()
+	defer teardown(server)
+
+	mux.HandleFunc("/api/v4/projects/1/approvals", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "POST")
+		testBody(t, r, `{"approvals_before_merge":3}`)
```

Comment:
I think this is the only line of the test that actually test something. The rest of this function only tests what is defined in this same file. So I wonder if we should add these tests?

Reply:
Yeah I get that, but both the raw JSON and the type are defined in this package and so if the API changes, our test will still pass but actual calls would fail.

I would leave out at least the `exampleParsedResponse` and add the raw JSON inside the method body of the test functions like done in other tests in this package. If you then specify just a handful of fields and check those fields in the returned response, it would be a good indication that the basic setup works.